storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -2178,6 +2281,18 @@ dberr_t dict_load_tables_from_space_id(space_id_t space_id, THD *thd,
     goto error;
   }
 
+  /* Before 8.0.24 if the table is used in EXCHANGE PARTITION or IMPORT. Even
+  after upgrade to the latest version 8.0.25 (which fixed the duplicate SDI
+  issue), such tables continue to contain duplicate SDI. PXB will scan the DD
+  table "mysql.tables" to determine the correct SDI */
+  if (ib_vector.sdi_vector->m_vec.size() > 2 &&
+      strcmp(space->name, "mysql") != 0 &&
+      fsp_is_file_per_table(space_id, space->flags)) {
+    msg("xtrabackup: duplicate sdi for space %s.\n", space->name);
+
+    get_id_from_mysql_table_scan(space->name, &sdi_id, thd);
```

Comment:
There is a way to avoid this DD scan can multiple times. Idea is something like this
1. save the list of all tables with duplicate SDI (table_names, current sdi_ids, etc)
2. after the normal IBDs are processed, if there are tables with duplicate SDI (saved_list not empty)
    we now start DD scan
3. for every DD record, we match from the saved_list and return correct sdi_ids
4. for the processed duplicate SDI tablespace, we remove the saved_list
5. when either saved_list is empty or when we reach the end of DD we stop

Reply:
building the map of <schema id, name> and key during first scan of duplicate sdi